### PR TITLE
enh(API): Interfaces for update security policy use case

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

This PR add the interfaces structures and test for the use case find security policy.

**Fixes** MON-11965

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: src/Core/Application/Security/UseCase/UpdateSecurityPolicy/UpdateSecurityPolicy.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    /**
+     * @param UpdateSecurityPolicyPresenterInterface $presenter
+     * @param UpdateSecurityPolicyRequest $request
+     */
+    public function __invoke(
+        UpdateSecurityPolicyPresenterInterface $presenter,
+        UpdateSecurityPolicyRequest $request
+    ): void {
+        $this->repository->updateSecurityPolicy(SecurityPolicyFactory::createFromRequest($request));
+        $presenter->setResponseStatus(new UpdateSecurityPolicyNoContentResponse());
```
Comment (a-launois) [inline src/Core/Application/Security/UseCase/UpdateSecurityPolicy/UpdateSecurityPolicy.php:49]:
```suggestion
        $presenter->setResponseStatus(new UpdateSecurityPolicyResponse());
```

File: src/Core/Application/Security/UseCase/UpdateSecurityPolicy/UpdateSecurityPolicyNoContentResponse.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Application\Security\UseCase\UpdateSecurityPolicy;
+
+use Core\Application\Common\UseCase\NoContentResponse;
+
+class UpdateSecurityPolicyNoContentResponse extends NoContentResponse
```
[the remark this answers is not in the document]
Comment (jeremyjaouen) [reply src/Core/Application/Security/UseCase/UpdateSecurityPolicy/UpdateSecurityPolicyNoContentResponse.php:28]: the class is named `UpdateSecurityPolicyNoContentResponse` to handle case were we also could have a `UpdateSecurityPolicyErrorResponse` etc, I think its better to name the class with his complete intention, as I have done [here](https://github.com/centreon/centreon/blob/password-security-policy/src/Core/Application/Security/UseCase/FindSecurityPolicy/FindSecurityPolicyErrorResponse.php)

[the remark this answers is not in the document]
Comment (callapa) [reply src/Core/Application/Security/UseCase/UpdateSecurityPolicy/UpdateSecurityPolicyNoContentResponse.php:28]: There is no need to create a new class, just use the NoContentResponse class.

[the remark this answers is not in the document]
Comment (jeremyjaouen) [reply src/Core/Application/Security/UseCase/UpdateSecurityPolicy/UpdateSecurityPolicyNoContentResponse.php:28]: if I use NoContentResponse directly, the condition in the JsonPresenter will not work (`is_subclass_of($this->data, NoContentResponse::class, false)`).

[the remark this answers is not in the document]
Comment (callapa) [reply src/Core/Application/Security/UseCase/UpdateSecurityPolicy/UpdateSecurityPolicyNoContentResponse.php:28]:
```suggestion
(is_subclass_of($this->data, NoContentResponse::class, false) || get_class($this->data) === NoContentResponse::class)
```

[the remark this answers is not in the document]
Comment (callapa) [reply src/Core/Application/Security/UseCase/UpdateSecurityPolicy/UpdateSecurityPolicyNoContentResponse.php:28]: (is_subclass_of($this->data, NoContentResponse::class, false) || get_class($this->data) === NoContentResponse::class)

**Review by callapa — APPROVED** [review]

File: tests/rest_api/rest_api.postman_collection.json [hunk 3]
```diff
@@ -12122,7 +12122,7 @@
 						],
 						"body": {
 							"mode": "raw",
-							"raw": "{\n  \"action\": \"add\",\n  \"object\": \"contacttpl\",\n  \"values\": \"test_name;test_alias;test@localhost;pwd;0;0;en_US;Idap\"\n}"
+							"raw": "{\n  \"action\": \"add\",\n  \"object\": \"contacttpl\",\n  \"values\": \"test_name;test_alias;test@localhost;Centreon@2022;0;0;en_US;Idap\"\n}"
```
Comment (a-launois) [inline tests/rest_api/rest_api.postman_collection.json:12125]:
```suggestion
							"raw": "{\n  \"action\": \"add\",\n  \"object\": \"contacttpl\",\n  \"values\": \"test_name;test_alias;test@localhost;Centreon@2022;0;0;en_US;ldap\"\n}"
```

File: tests/rest_api/rest_api.postman_collection.json [hunk 4]
```diff
@@ -12986,7 +12986,7 @@
 						],
 						"body": {
 							"mode": "raw",
-							"raw": "{\n  \"action\": \"add\",\n  \"object\": \"contacttpl\",\n  \"values\": \"{{ctpl_name2}};{{ctpl_alias2}};test@localhost;pwd;0;0;en_US;Idap\"\n}"
+							"raw": "{\n  \"action\": \"add\",\n  \"object\": \"contacttpl\",\n  \"values\": \"{{ctpl_name2}};{{ctpl_alias2}};test@localhost;Centreon@2022;0;0;en_US;Idap\"\n}"
```
Comment (a-launois) [inline tests/rest_api/rest_api.postman_collection.json:12989]:
```suggestion
							"raw": "{\n  \"action\": \"add\",\n  \"object\": \"contacttpl\",\n  \"values\": \"{{ctpl_name2}};{{ctpl_alias2}};test@localhost;Centreon@2022;0;0;en_US;ldap\"\n}"
```

File: tests/rest_api/rest_api.postman_collection.json [hunk 5]
```diff
@@ -14795,7 +14795,7 @@
 						],
 						"body": {
 							"mode": "raw",
-							"raw": "{\n  \"action\": \"add\",\n  \"object\": \"contact\",\n  \"values\": \"{{contact_name2}};{{contact_alias2}};test@localhost;pwd;0;0;en_US;Idap\"\n}"
+							"raw": "{\n  \"action\": \"add\",\n  \"object\": \"contact\",\n  \"values\": \"{{contact_name2}};{{contact_alias2}};test@localhost;Centreon@2022;0;0;en_US;Idap\"\n}"
```
Comment (a-launois) [inline tests/rest_api/rest_api.postman_collection.json:14798]:
```suggestion
							"raw": "{\n  \"action\": \"add\",\n  \"object\": \"contact\",\n  \"values\": \"{{contact_name2}};{{contact_alias2}};test@localhost;Centreon@2022;0;0;en_US;ldap\"\n}"
```

**Review by kduret — APPROVED** [review]

**Pull request merged**